lib/watcher.js

```diff
@@ -24,75 +28,65 @@ class Watcher {
-          }
-          // Since multiple changes can occur at the same time, lets wait a bit before processing.
-          if (this.fsWait) return;
-          this.fsWait = setTimeout(async () => {
-            await changeCallback(this.filesChanged);
-            this.filesChanged = {};
-            this.fsWait = false;
-            this.watch(changeCallback, errorCallback);
-          }, 500);
-        } catch (e) {
-          // File was not found find all that are missing..
-          const unknownPaths = this.getAllNonExistingPaths();
-          if (unknownPaths.length > 0) {
-            this.closeWatchers();
-            errorCallback(unknownPaths);
-          }
-        }
-      });
+      const watcher = chokidar.watch(path, {ignoreInitial: true});
+      watcher.on('all', (eventType, changedPath) => this.fileChanged(path, changedPath, eventType, changeCallback, errorCallback));
       this.watchers[path] = watcher;
     }
   }
 
   /**
-   * Convert the event type to a more usefull one.
-   * @param {*} currentEventType
-   * @param {*} path
-   * @param {*} fileName
+   * Should be called when a file has changed one way or another.
+   * @param {*} listenerPath The path the watcher is listening on.
+   * @param {*} changedPath The file/dir that was changed
+   * @param {*} eventType What kind of change
+   * @param {*} changeCallback Callback to call when changed occur.
+   * @param {*} errorCallback Calback to call when it is no longer possible to watch a file.
    */
-  convertEventType(currentEventType, path, fileName) {
-    let newEventType = currentEventType;
-    const stat = fs.statSync(path);
-    if (stat.isDirectory()) {
-      const filePath = Path.resolve(path, `./${fileName}`);
-      if (!fs.existsSync(filePath)) {
-        newEventType = 'removed';
-      } else if (currentEventType === 'change') {
-        const filePathStat = fs.statSync(filePath);
-        //Not all unix have access to birthtime
-        if (filePathStat.isFile() && filePathStat.birthtimeMs) {
-          const milliseconds = (new Date().getTime() - filePathStat.birthtimeMs);
-          if (milliseconds < 500) {
-            newEventType = 'added';
-          }
-        } else {
-          return null;
-        }
-      } else if (currentEventType === 'rename') {
-        newEventType = 'added';
+  fileChanged(listenerPath, changedPath, eventType, changeCallback, errorCallback) {
+    try {
+      if (fs.existsSync(listenerPath)) {
+        const newEventType = this.convertEventType(eventType);
+        this.filesChanged[changedPath] = { eventType: newEventType, path: changedPath};
+        // Since multiple changes can occur at the same time, lets wait a bit before processing.
+        if (this.fsWait) return;
+        this.fsWait = setTimeout(async () => {
+          await changeCallback(this.filesChanged);
+          this.filesChanged = {};
+          this.fsWait = false;
+        }, 500);
```

Comment:
Should we make this time configurable?

Reply:
Not sure what people could use it for :smile: This timer is to ensure that all files has been processed by the watcher before actually calling the callback. Else it would re-generate for each file that was changed.. Might have to refactor it a bit so it keeps waiting as long as there are changes and then refresh the timeout instead of waiting right after the first change was caught. Not sure

Reply:
Ohh, I see what you mean. Totally misunderstood the purpose. Nevermind 👍